src/analysis/calculus/cont_diff.lean

```diff
@@ -2602,3 +2617,53 @@ begin
 end
 
 end
+
+section apply
+
+lemma norm_iterated_fderiv_within_clm_apply {f : E → (F →L[𝕜] G)} {g : E → F} {s : set E} {x : E}
+  {N : ℕ∞} {n : ℕ} (hf : cont_diff_on 𝕜 N f s) (hg : cont_diff_on 𝕜 N g s) (hs : unique_diff_on 𝕜 s)
+  (hx : x ∈ s) (hn : ↑n ≤ N) :
+    ‖iterated_fderiv_within 𝕜 n (λ y, (f y) (g y)) s x‖ ≤
+    (finset.range (n + 1)).sum (λ i, ↑(n.choose i) * ‖iterated_fderiv_within 𝕜 i f s x‖ *
+      ‖iterated_fderiv_within 𝕜 (n - i) g s x‖) :=
+begin
+  let B : (F →L[𝕜] G) →L[𝕜] F →L[𝕜] G :=
+  continuous_linear_map.flip (continuous_linear_map.apply 𝕜 G),
+  have hB : ‖B‖ ≤ 1 :=
+  begin
+    simp only [continuous_linear_map.op_norm_flip, continuous_linear_map.apply],
+    refine continuous_linear_map.op_norm_le_bound _ zero_le_one (λ f, _),
+    simp only [continuous_linear_map.coe_id', id.def, one_mul],
+  end,
+  refine (B.norm_iterated_fderiv_within_le_of_bilinear hf hg hs hx hn).trans _,
+  exact le_of_le_of_eq (mul_le_mul_of_nonneg_right hB (finset.sum_nonneg (λ _ _, by positivity)))
+    (one_mul _),
+end
+
+lemma norm_iterated_fderiv_clm_apply {f : E → (F →L[𝕜] G)} {g : E → F}
+  {N : ℕ∞} {n : ℕ} (hf : cont_diff 𝕜 N f) (hg : cont_diff 𝕜 N g) (x : E) (hn : ↑n ≤ N):
+    ‖iterated_fderiv 𝕜 n (λ (y : E), (f y) (g y)) x‖ ≤
+      (finset.range (n + 1)).sum (λ (i : ℕ), ↑(n.choose i) * ‖iterated_fderiv 𝕜 i f x‖ *
+        ‖iterated_fderiv 𝕜 (n - i) g x‖) :=
+begin
+  simp_rw [← iterated_fderiv_within_univ],
+  exact norm_iterated_fderiv_within_clm_apply hf.cont_diff_on hg.cont_diff_on unique_diff_on_univ
+    (set.mem_univ x) hn,
+end
+
+lemma norm_iterated_fderiv_clm_apply_const {f : E → (F →L[𝕜] G)} {c : F} {x : E}
+  {N : ℕ∞} {n : ℕ} (hf : cont_diff 𝕜 N f) (hn : ↑n ≤ N) :
+    ‖iterated_fderiv 𝕜 n (λ (y : E), (f y) c) x‖ ≤ ‖c‖ * ‖iterated_fderiv 𝕜 n f x‖ :=
+begin
+  rw [mul_comm],
```

Comment:
This looks terribly complicated to use a result on the iterated derivative of bilinear maps to deduce something on the iterated derivative of composition with a linear map. Is `continuous_linear_map.iterated_fderiv_comp_left` (or the neighboring lemmas) enough for you? Otherwise, maybe add a few lemmas there to make things smoother.

Reply:
I agree that this overly complicated, but I did not find a better way to prove it. I am not understanding what are telling me. We have a (nonlinear) map into the space of continuous linear maps here, so I don't how this should follow from `continuous_linear_map.iterated_fderiv_comp_left`.

Reply:
The map `G` which to a linear map `g` associates `g c` is itself a linear map, and you want to bound the iterated derivative of `G o f`. For this, `continuous_linear_map.iterated_fderiv_comp_left` (or a variant of it) should apply, right?